client/src/connection/rest/index.ts

```diff
@@ -289,9 +289,11 @@ class RestSession extends Session {
    * @param session a session id to print logs for.
    */
   private printSessionLog = async (session: ComputeSession) => {
-    const logs = await session.getLogStream();
-    for await (const log of logs) {
-      this._onLogFn(log);
+    if (this._onLogFn) {
```

Comment:
I'm thinking it would be good if we can make this kind of situation more safe.
If we would like to always show log, how about make the `LogChannelFn` the default one for base Session?
If we don't want a default implementation, how about declare the type of the `_onLogFn` to be `OnLogFn | undefined` so that every usage knows that it could be undefined.

Reply:
Good point, once we know what a desired UX is I can either update the type or move it to be base behavior.

Reply:
@scnwwu if its agreeable I'd like to implement option#2. If we have further feature areas in the future that would require it then we could refactor to move it to the default implementation?